### PR TITLE
add week2

### PR DESCRIPTION
**Pull request by LouisaStrunk02**
Changes:
1. add scrabble
2. add readability
3. add caesar
4. add substitution

### CODE REVIEW

File: week2/caesar/caesar.c [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    for (int i = 0, length = strlen(keyinput); i < length; i++)
+    {
+        if (!isdigit(keyinput[i]))
+        {
+            return false;
+        }
+    }
+}
+
+// char rotate(char text[], int k)
```
Comment (DevBastiS) [inline week2/caesar/caesar.c:61]: Remove unused Codes.

**Review by DevBastiS — APPROVED** [review]

File: week2/readability/readability.c [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    for (int i = 0, length = strlen(text); i < length; i++)
+    {
+        if (text[i] == ' ')
+        {
+            numberofwords += 1;
+        }
+        else
+        {
+            numberofwords = numberofwords;
+        }
```
Comment (MKrips) [inline week2/readability/readability.c:65]:
```suggestion

```
This code has no real function, so you dont need it.

File: week2/readability/readability.c [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    for (int i = 0, length = strlen(text); i < length; i++)
+    {
+        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
+        {
+            numberofsentences += 1;
+        }
+        else
+        {
+            numberofsentences = numberofsentences;
+        }
```
Comment (MKrips) [inline week2/readability/readability.c:84]:
```suggestion

```
Same as above

File: week2/readability/readability.c [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    }
+}
+
+int count_letters(string text)
+{
+    int numberofletters = 0;
+
+    for (int i = 0, length = strlen(text); i < length; i++)
+    {
+        if (text[i] >= 'A' && text[i] <= 'Z' || text[i] >= 'a' && text[i] <= 'z')
```
Comment (MKrips) [inline week2/readability/readability.c:43]: As far as I am aware, the library you use has an isalpha-function that checks whether you have a letter. That function would be better here for readability purposes.

File: week2/substitution/substitution.c [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+    }
+
+    string key = argv[1];
+    string plaintext = get_string("Plaintext: ");
+
+    for (int i = 0, length = strlen(key); i < length; i++)
+    {
+        if (islower(key[i]))
+        {
+            key[i] = key[i] - 32;
```
Comment (MKrips) [inline week2/substitution/substitution.c:57]:
```suggestion
            key[i] = toupper(key[i]);
```
This is more readable, provided you want everything uppercase.

**Review by MKrips — CHANGES_REQUESTED** [review]
Just some readability changes. Dont forget that the cs50-library has some neat functions that help make your code more readable.

**Review by MKrips — APPROVED** [review]

**Pull request merged**